tests/models/test_headers.py

```diff
@@ -23,7 +23,8 @@ def test_headers():
     assert dict(h) == {"a": "123, 456", "b": "789"}
     assert repr(h) == "Headers([('a', '123'), ('a', '456'), ('b', '789')])"
     assert h == httpx.Headers([("a", "123"), ("b", "789"), ("a", "456")])
-    assert h != [("a", "123"), ("A", "456"), ("b", "789")]
+    assert h == [("a", "123"), ("A", "456"), ("b", "789")]
+    assert h == {"a": "123", "A": "456", "b": "789"}
```

Comment:
To keep coverage at 100% we'll also need a test against a no-headers-like item.

Eg...

```python
assert h != "a: 123\nA: 456\nb: 789"
```